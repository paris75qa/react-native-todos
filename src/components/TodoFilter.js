import React, {PropTypes} from 'react';
import { View, Text, TouchableHighlight, StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  filter: {
    flex: 0,
    flexDirection: "row",
  },
  item: {
    paddingLeft: 10,
    paddingRight: 10,
    backgroundColor: "#CCC",
  },
  selected: {
    backgroundColor: "#EEE",
  }
});

const TodoFilter = (props) => {

  const allStyle = (props.status==="none") ? styles.selected : "";
  const completeStyle = (props.status==="complete") ? styles.selected : "";
  const todoClass = (props.status==="todo") ? styles.selected : "";

  return <View>
      <Text>Filtres: </Text>
    <View style={styles.filter}>
      <TouchableHighlight style={[styles.item, allStyle]} onPress={props.onSelectFilter.bind(this,"none")}><Text>Tous</Text></TouchableHighlight>
      <TouchableHighlight style={[styles.item, completeStyle]} onPress={props.onSelectFilter.bind(this,"complete")}><Text>Termin√©</Text></TouchableHighlight>
      <TouchableHighlight style={[styles.item, todoClass]} onPress={props.onSelectFilter.bind(this,"todo")}><Text>A faire</Text></TouchableHighlight>
    </View>
  </View>
}

TodoFilter.propTypes = {
  status: PropTypes.string.isRequired,
  onSelectFilter: PropTypes.func.isRequired,
}

export default TodoFilter;
