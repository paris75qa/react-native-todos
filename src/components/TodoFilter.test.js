import React from 'react';
import TodoFilter from './TodoFilter';
import renderer from 'react-test-renderer';

it('TodoFilter status="none" renders correctly', () => {
  const tree = renderer.create(
    <TodoFilter status="none" onSelectFilter={()=>{}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TodoFilter status="complete" renders correctly', () => {
  const tree = renderer.create(
    <TodoFilter status="complete" onSelectFilter={()=>{}}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('TodoFilter status="todo" renders correctly', () => {
  const tree = renderer.create(
    <TodoFilter status="todo" onSelectFilter={()=>{}}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
