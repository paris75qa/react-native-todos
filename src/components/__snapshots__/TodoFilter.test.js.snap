// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilter status="complete" renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Filtres: 
  </Text>
  <View
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "",
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Tous
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            Object {
              "backgroundColor": "#EEE",
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Terminé
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "",
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        A faire
      </Text>
    </View>
  </View>
</View>
`;

exports[`TodoFilter status="none" renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Filtres: 
  </Text>
  <View
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            Object {
              "backgroundColor": "#EEE",
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Tous
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "",
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Terminé
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "",
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        A faire
      </Text>
    </View>
  </View>
</View>
`;

exports[`TodoFilter status="todo" renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Filtres: 
  </Text>
  <View
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "",
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Tous
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            "",
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Terminé
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "backgroundColor": "#CCC",
              "paddingLeft": 10,
              "paddingRight": 10,
            },
            Object {
              "backgroundColor": "#EEE",
            },
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        A faire
      </Text>
    </View>
  </View>
</View>
`;
